pkg/vsphere/disk/disk.go

```diff
@@ -45,6 +45,43 @@ type Filesystem interface {
 	Unmount(path string) error
 }
 
+// ReferenceCount represents the number of references to a disk
+type ReferenceCount struct {
+	owner   string
+	refname string
+	count   int
+}
+
+// NewReferenceCount creates and returns a ReferenceCount initialized to 0
+func NewReferenceCount(o, n string) *ReferenceCount {
+	return &ReferenceCount{
+		owner:   o,
```

Comment:
Owner is confusing when applied to locking structures - this is not the owner of a lock, it's the identifier of the resource.
Different name please.